src/main/java/at/hannibal2/skyhanni/features/inventory/caketracker/CakeTracker.kt

```diff
@@ -0,0 +1,374 @@
+import at.hannibal2.skyhanni.features.inventory.patternGroup
+import at.hannibal2.skyhanni.skyhannimodule.SkyHanniModule
+import at.hannibal2.skyhanni.utils.CollectionUtils.addSearchString
+import at.hannibal2.skyhanni.utils.ColorUtils.toChromaColor
+import at.hannibal2.skyhanni.utils.HypixelCommands
+import at.hannibal2.skyhanni.utils.InventoryUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.NumberUtil.formatInt
+import at.hannibal2.skyhanni.utils.RegexUtils.groupOrNull
+import at.hannibal2.skyhanni.utils.RegexUtils.matchMatcher
+import at.hannibal2.skyhanni.utils.RegexUtils.matches
+import at.hannibal2.skyhanni.utils.RenderUtils.highlight
+import at.hannibal2.skyhanni.utils.SimpleTimeMark
+import at.hannibal2.skyhanni.utils.SkyBlockTime
+import at.hannibal2.skyhanni.utils.renderables.Renderable
+import at.hannibal2.skyhanni.utils.renderables.Searchable
+import at.hannibal2.skyhanni.utils.renderables.toSearchable
+import at.hannibal2.skyhanni.utils.tracker.SkyHanniTracker
+import at.hannibal2.skyhanni.utils.tracker.TrackerData
+import com.google.gson.annotations.Expose
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+import kotlin.time.Duration.Companion.milliseconds
+
+@SkyHanniModule
+object CakeTracker {
+
+    private fun getCakeTrackerData() = ProfileStorageData.profileSpecific?.cakeTracker
+    private val config get() = SkyHanniMod.feature.inventory.cakeTracker
+    private var currentYear = 0
+
+    private var inCakeBag = false
+    private var inCakeInventory = false
+    private var timeOpenedCakeInventory = SimpleTimeMark.farPast()
+    private var inAuctionHouse = false
+    private var unobtainedCakesDisplayed = false
+    private var searchingForCakes = false
+    private var knownCakesInCurrentInventory = mutableListOf<Int>()
+
+    /**
+     * REGEX-TEST: §cNew Year Cake (Year 360)
+     * REGEX-TEST: §cNew Year Cake (Year 1,000)
+     * REGEX-TEST: §f§f§cNew Year Cake (Year 330)
+     */
+    private val cakeNamePattern by patternGroup.pattern(
+        "cake.name",
+        "(?:§f§f)?§cNew Year Cake \\(Year (?<year>[\\d,]*)\\)",
+    )
+
+    /**
+     * REGEX-TEST: Ender Chest (2/9)
+     * REGEX-TEST: Jumbo Backpack (Slot #6)
+     * REGEX-TEST: New Year Cake Bag
+     */
+    private val cakeContainerPattern by patternGroup.pattern(
+        "cake.container",
+        "^(Ender Chest \\(\\d{1,2}/\\d{1,2}\\)|.*Backpack(?:§r)? \\(Slot #\\d{1,2}\\)|New Year Cake Bag)$",
+    )
+
+    /**
+     * REGEX-TEST: New Year Cake Bag
+     */
+    private val cakeBagPattern by patternGroup.pattern(
+        "cake.bag",
+        "^New Year Cake Bag$",
+    )
+
+    /**
+     * REGEX-TEST: Auctions Browser
+     * REGEX-TEST: Auctions: "Test"
+     */
+    private val auctionBrowserPattern by patternGroup.pattern(
+        "auction.search",
+        "^(?:Auctions Browser|Auctions: \".*)$",
```

Comment:
if you arent using any find/findmatchers anywhere, using `^$` in your regex is useless iirc